src/command_modules/azure-cli-billing/azure/cli/command_modules/billing/custom.py

```diff
@@ -0,0 +1,20 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def cli_billing_list_invoices(client, generate_url=False):
+    '''List all available invoices of the subscription'''
+    invoices = client.list(expand='downloadUrl' if generate_url else None)
+    return list(invoices)
+
+
+def cli_billing_get_invoice(client, name):
+    '''Retrieve invoice of specific name of the subscription'''
+    return client.get(name)
+
+
+def cli_billing_get_billing_period(client, name):
+    '''Retrieve billing period of specific name of the subscription'''
+    return client.get(name)
```

Comment:
This should probably be removed and the commands.py file updated to do the pure reflection on the SDK.